<launch>
	<!--param name="use_sim_time" value="true" /-->
   <arg name="manager" default="pointcloud_manager" />

   <!--node pkg="nodelet" type="nodelet" name="$(arg manager)" args="manager" output="screen" /-->
  
  <!--node pkg="nodelet" type="nodelet" name="depth_image_registration"
        args="load depth_image_proc/register $(arg manager)"
        respawn="false">
    <remap from="rgb/camera_info"             to="/camera/rgb/camera_info" />
    <remap from="depth/camera_info"           to="/camera/depth/camera_info" />
    <remap from="depth/image_rect"            to="/camera/depth/image" />
    
    <remap from="depth_registered/camera_info"    to="/camera/depth_registered/camera_info" />
    <remap from="depth_registered/image_rect"     to="/camera/depth_registered/image_raw" />
  </node-->

  <!--node pkg="nodelet" type="nodelet" name="points_xyzrgb_generator"
        args="load depth_image_proc/point_cloud_xyzrgb $(arg manager)"
        respawn="false">
    <remap from="rgb/image_rect_color"        to="/camera/rgb/image_color" />
    <remap from="rgb/camera_info"             to="/camera/rgb/camera_info" />
    <remap from="depth_registered/image_rect" to="/camera/depth/image" />
    <remap from="depth_registered/points"     to="/camera/depth_registered/points" />
    <param name="queue_size" value="20" />
  </node-->
    <!--remap from="depth_registered/image_rect" to="/camera/depth_registered/image_raw" /-->
	
	<node pkg="crosbot_3d_position_track_full" type="crosbot_3d_position_track_full" name="crosbot_3d_position_track_full" output="screen">
      <param name="icp_frame" value="world" />
      <param name="base_frame" value="kinect" />
      <param name="ImageWidth" value="640" />
      <param name="ImageHeight" value="480" />
      <param name="rgb_sub" value="/camera/rgb/image_color" />
      <param name="depth_sub" value="/camera/depth/image" />
      <param name="OutputCurrentMap" value="true" />      
      <param name="PublishTransform" value="true" />
      <param name="PublishMessage" value="true" />
      <param name="SkipNumCheckBlocks" value="4" />
      <param name="TruncNeg" value="0.2" />
      <param name="TruncPos" value="0.3" />
      <param name="LocalMapWidth" value="8" />
      <param name="LocalMapHeight" value="4" />
      <param name="CellSize" value="0.0125" />
      <param name="NumBlocksAllocated" value="13000" />
      <param name="MaxPointsFrac" value="5" />
      <param name="SliceMult" value="1" />
      <param name="OutputCurrentMapRate" value="3000000" />
      <param name="DistThresh" value="0.01" />
      <param name="DistThresh2" value="0.2" />
      <param name="DistThresh4" value="0.3" />
      <param name="MaxMoveXYZ" value="0.8" />
      <param name="MaxMoveRPY" value="0.7" />
      <param name="MinICPCount" value="800" />
      <param name="MinScale" value="2000" />
      <param name="ScaleRegularisation" value="0" />
      <param name="MaxRayTrace" value="40" />
      <param name="ReExtractBlocks" value="true" />
      <param name="UseICP" value="true" />
      <param name="UseOdometry" value="false" />

	</node>
   
   <node pkg="crosbot_3d_graphslam_display" type="crosbot_3d_graphslam_display" name="crosbot_3d_graphslam_display" output="screen" launch-prefix="nice -n 10">
      <param name="CreateMesh" value ="true" />
      <param name="UseVisualiser" value ="false" />
      <param name="WarpMaps" value ="false" />
   </node>
   

</launch>	
